### PR TITLE
Add a compilation option to disable topology

### PR DESCRIPTION
**Pull request by ufownl**
I ran the latest version of gemma.cpp on one of our devices, which has the topology `16x1x2, using 16x1x2`. The running speed was far slower than the previous versions (with a single thread pool). Therefore, I made this PR to add an option to disable topology detection and create a single package (with one cluster and one node) in `NestedPools` to run it with the same performance as before on these devices.

### CODE REVIEW

File: util/threading.h [hunk 1]
```diff
@@ -104,6 +108,7 @@ class BoundedTopology {
       enabled_lps.Set(lp);
     }
 
+#if !GEMMA_DISABLE_TOPOLOGY
     if (HWY_LIKELY(!topology_.packages.empty())) {
       InitFromTopology(enabled_lps, package_slice, cluster_slice);
     }
```
Comment (jan-wassenberg) [inline util/threading.h:114]: I think we could just put an `#endif` here - without InitFromTopology, we will not have filled packages_, which is what NumPackages checks, so we can use the same fallback.

**Review by jan-wassenberg — COMMENTED** [review]
Thanks, I understand the desire for this change. FYI we will soon be updating the matmul to actually use these clusters and restore or hopefully further improve performance vs before. But I see no harm in keeping this option.

**Comment by ufownl** [comment]
Now, it uses that fallback. :)

**Review by jan-wassenberg — APPROVED** [review]
Nice, thanks for making the change :)

**Pull request merged**